core/email.go

```diff
@@ -16,22 +16,26 @@ var (
 	ErrEmailAddressEmpty   = errors.New("e-mail address is empty")
 )
 
-type EmailAddress interface {
-	// String returns the string representation for this e-mail address
-	String() string
-}
-
-type emailAddress struct {
+type EmailAddress struct {
 	address string
 }
 
-func (email emailAddress) String() string {
+func (email *EmailAddress) String() string {
 	return email.address
 }
 
-// NewEmailAddress parses an e-mail address from any string.
+func (email *EmailAddress) UnmarshalText(text []byte) error {
+	add, err := ParseEmailAddress(string(text))
+	if err != nil {
+		return err
+	}
+	email.address = add.address
```

Comment:
Het komt op hetzelfde neer in de praktijk, maar ik denk dat het misschien mooier zou zijn om hier ook `*email = add` te doen? Dat maakt het iets gemakkelijker om dit te copy-pasten naar meer complexe data types (wat sowieso ooit gaat gebeuren afgaande op hoeveel andere boilerplate er continu gekopieerd wordt 😛)

Reply:
Dat gaat denk ik niet. `email` is al een pointer. Het is net op die pointer dat je het adress field wil aanpassen. Qua types zou je `email = add` kunnen doen, maar dat is een 'ineffectual assignment' voor de linter

Reply:
Is dat niet net het punt van een UnmarshalText: je stelt de velden in op het huidige object?

Reply:
Ja, dat weet ik `*email` dereferencet een pointer, dat zou niet werken moest email niet al een pointer zijn 😛 
`email = add` is een ineffectual assignment omdat je dan enkel de lokale pointer aanpast maar niet het object waar die naar verwijst (wat `*email = add` wel doet).  
Dat stelt ook de velden in op het huidige object, dat is wat ik bedoel: wat je nu doet is exact hetzelfde als `*email = add`, alleen moet je de velden apart opsommen. Op andere plaatsen gebruik je wel een copy van de hele struct, dat is wat `*email = add` zou doen